R/prepare_cdt.R

```diff
@@ -0,0 +1,73 @@
+#' Prepare CDT
+#'
+#' @param data The data.frame to calculate from.
+#' @param date The name of the date column in \code{data}.
+#' @param station The name of the station column in \code{data}.
+#' @param year The name of the year column in \code{data}.
+#' @param element The name of the element column in \code{data}.
+#' @param metadata The metadata data.frame.
+#' @param latitude The name of the latitude column in \code{metadata}.
+#' @param longitude The name of the longitude column in \code{metadata}.
+#' @param altitude The name of the altitude column in \code{metadata}.
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' # create summary data
+#' data("daily_niger"); data("stations_niger")
+#' summary_data <- daily_niger %>%
+#' dplyr::mutate(dekad_date = dekad(daily_niger$date)) %>%
+#' dplyr::group_by(station_name, year, dekad_date) %>%
+#' dplyr::summarise(date = dplyr::first(date), sum = sum(tmax))
+#' 
+#' prepare_cdt(data = summary_data, date = "date", year = "year",
+#'             station = "station_name",
+#'             element = "sum", metadata = stations_niger, 
+#'             latitude = "lat", longitude = "long", altitude = "alt")
+
+
+prepare_cdt <- function(data, date, year, element, metadata = NULL,
+                        station, latitude, longitude, altitude) {
+  checkmate::assert_data_frame(data)
+  checkmate::assert_string(year)
+  assert_column_names(data, year)
+  assert_column_names(data, date)
+  checkmate::assert_string(element)
+  assert_column_names(data, element)
+  checkmate::assert_data_frame(metadata, null.ok = TRUE)
+  checkmate::assert_string(station)
+  checkmate::assert_string(latitude)
+  checkmate::assert_string(longitude)
+  checkmate::assert_string(altitude)
+  # data_with_meta is whichever data.frame has the metadata columns in
+  if (is.null(metadata)) {
+    data_with_meta <- data
+  } else {
+    data_with_meta <- metadata
```

Comment:
Sorry, I didn't mean that there should not be any joining if `metadata` is supplied. If `metadata` is given then it means lon and lat are in metadata but elements data is in `data`, so they need to be joined to get them into one, as is done in geoclim prepare.

Reply:
Yes I saw that. But in the CDT format they want the two data frames appended rather than merged

Reply:
Ah yes, I was confused by my own code from geoclim! So ignore that!